File: src/main/java/HyperCore/HyperMonster/HyperCreeper.java
package HyperCore.HyperMonster;

import HyperCore.HyperCore;
import HyperCore.Listener.Hyper;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.entity.Creeper;
import org.bukkit.event.EventHandler;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.event.entity.EntityExplodeEvent;
import org.bukkit.event.entity.EntitySpawnEvent;
import org.bukkit.scheduler.BukkitRunnable;

public class HyperCreeper extends Hyper {

    @EventHandler
    public void onCreeperExplode(EntityExplodeEvent event) {
        if (event.getEntity() instanceof Creeper creeper) {
            Location center = creeper.getLocation();
            World world = center.getWorld();
            if (world == null) return;

            new BukkitRunnable() {
                int ticks = 0;
                @Override
                public void run() {
                    if (ticks++ >= 100) {
                        this.cancel();
                        return;
                    }
                    double radiusPerTick = 0.5;
                    double pointPerCircum = 6.0;
                    double radius = radiusPerTick * ticks;
                    double circum = 2 * Math.PI * radius;
                    int pointsCount = (int) (circum / pointPerCircum);
                    if (pointsCount == 0) return;

                    double angle = 360.0 / pointsCount;
                    double y = center.getY();

                    for (int i = 0; i < pointsCount; i++) {
                        double currentAngle = Math.toRadians(i * angle);
                        double x = -Math.sin(currentAngle);
                        double z = Math.cos(currentAngle);

                        world.createExplosion(center.getX() + x * radius, y, center.getZ() + z * radius, 4F, false, true);
                    }
                }
            }.runTaskTimer(HyperCore.getInstance(), 0L, 1L);
        }
    }

    @EventHandler
    public void onHit(EntityDamageByEntityEvent event) {
        if (event.getEntity() instanceof Creeper) {
            event.setDamage(0.0);
        }
    }

    @EventHandler
    public void onSpawn(EntitySpawnEvent event) {
        if (event.getEntity() instanceof Creeper creeper) {
            creeper.setPowered(true);
        }
    }
}

File: src/main/java/HyperCore/HyperMonster/HyperCreeper.kt
package HyperCore.HyperMonster

import HyperCore.Listener.Hyper
import org.bukkit.event.EventHandler
import org.bukkit.event.entity.EntityExplodeEvent
import kotlin.math.cos
import kotlin.math.sin

class HyperCreeper : Hyper() {

    val radius = 3.0

    val pointPerCircum = 6.0

    val circum = 2.0 * Math.PI * radius

    val pointsCount = (circum / pointPerCircum).toInt()

    @EventHandler
    fun onCreeperExplode(event: EntityExplodeEvent) {
        val entity = event.entity
        if (entity.type.name != "CREEPER") return

        val center = entity.location
        if (pointsCount == 0) return

        val angle = 360.0 / pointsCount
        val world = center.world
        val y = center.y

        for (i in 0 until pointsCount) {
            val currentAngle = Math.toRadians(i * angle)
            val x = -sin(currentAngle)
            val z = cos(currentAngle)

            world.createExplosion(center.x + x * radius, y, center.z + z * radius, 2.0f, false, true)
        }
    }

}
